core/commands/add.go

```diff
@@ -212,9 +218,13 @@ You can now check what blocks have been created by:
 			_, err = api.Unixfs().Add(req.Context, req.Files, opts...)
 		}()
 
-		err = res.Emit(events)
-		if err != nil {
-			return err
+		for event := range events {
+			output := event.(*coreiface.AddEvent)
```

Comment:
CoreAPI probably shouldn't expose string hashes, pushed a fix in https://github.com/ipfs/go-ipfs/pull/5854

Reply:
Blocked on https://github.com/ipfs/go-ipfs/pull/5854.

Reply:
@Stebalien I would ordinary preferred this the other way around, but since I said I was going to be OOO right now I'm fine with that, but could will get that in by the near year so this P.R. can move forward.

Reply:
Fair enough. I didn't realize this was a mostly unrelated issue.

(I merged #5854 anyways but I've rebased this PR at the same time)